src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC.js

```diff
@@ -0,0 +1,257 @@
+                    let constraintRespected = false;
+                    if (isFinite(nextWaypoint.legAltitude1) && altitude <= nextWaypoint.legAltitude1) {
+                        if (this._hasReachedTopOfDescent) {
+                            SimVar.SetSimVarValue("K:ALTITUDE_SLOT_INDEX_SET", "number", 2);
+                            Coherent.call("AP_ALT_VAR_SET_ENGLISH", 2, nextWaypoint.legAltitude1, true);
+                            SimVar.SetSimVarValue("L:AP_CURRENT_TARGET_ALTITUDE_IS_CONSTRAINT", "number", 1);
+                            constraintRespected = true;
+                        }
+                    }
+                    if (!constraintRespected) {
+                        SimVar.SetSimVarValue("K:ALTITUDE_SLOT_INDEX_SET", "number", 0);
+                        SimVar.SetSimVarValue("L:AP_CURRENT_TARGET_ALTITUDE_IS_CONSTRAINT", "number", 0);
+                    }
+                }
+                else {
+                    SimVar.SetSimVarValue("K:ALTITUDE_SLOT_INDEX_SET", "number", 0);
+                    SimVar.SetSimVarValue("L:AP_CURRENT_TARGET_ALTITUDE_IS_CONSTRAINT", "number", 0);
+                }
+            }
+            if (!this.flightPlanManager.isActiveApproach()) {
+                let activeWaypoint = this.flightPlanManager.getActiveWaypoint();
+                let nextActiveWaypoint = this.flightPlanManager.getNextActiveWaypoint();
+                if (activeWaypoint && nextActiveWaypoint) {
+                    let pathAngle = nextActiveWaypoint.bearingInFP - activeWaypoint.bearingInFP;
+                    while (pathAngle < 180) {
+                        pathAngle += 360;
+                    }
+                    while (pathAngle > 180) {
+                        pathAngle -= 360;
+                    }
+                    let absPathAngle = 180 - Math.abs(pathAngle);
+                    let airspeed = Simplane.getIndicatedSpeed();
+                    if (airspeed < 400) {
+                        let turnRadius = airspeed * 360 / (1091 * 0.36 / airspeed) / 3600 / 2 / Math.PI;
+                        let activateDistance = Math.pow(90 / absPathAngle, 1.6) * turnRadius * 1.2;
+                        let distanceToActive = Avionics.Utils.computeGreatCircleDistance(planeCoordinates, activeWaypoint.infos.coordinates);
+                        if (distanceToActive < activateDistance) {
+                            this.flightPlanManager.setActiveWaypointIndex(this.flightPlanManager.getActiveWaypointIndex() + 1);
+                        }
+                    }
+                }
+            }
+            SimVar.SetSimVarValue("SIMVAR_AUTOPILOT_AIRSPEED_MIN_CALCULATED", "knots", Simplane.getStallProtectionMinSpeed());
+            SimVar.SetSimVarValue("SIMVAR_AUTOPILOT_AIRSPEED_MAX_CALCULATED", "knots", Simplane.getMaxSpeed(Aircraft.CJ4));
+            this.updateAutopilotCooldown = this._apCooldown;
+        }
+    }
+
+    /**
```

Comment:
This is the meat of the changes, the rest of the file (unless indicated) is the stock Asobo code.